tests/applications/test_convergence_analysis.py

```diff
@@ -788,85 +792,201 @@ def grids() -> list[pp.Grid, pp.MortarGrid]:
+
+    return numerator / denominator
+
+
 @pytest.mark.parametrize("is_relative", [False, True])
 @pytest.mark.parametrize(
-    "is_sd, is_cc, is_scalar",
+    "is_sd, is_cc, is_scalar, parameter_weight",
     [
-        (True, True, True),  # subdomain scalar cell-centered quantity
-        (True, False, True),  # subdomain scalar face-centered quantity
-        (True, True, False),  # subdomain vector cell-centered quantity
-        (True, False, False),  # subdomain vector face-centered quantity
-        (False, True, True),  # interface scalar cell-centered quantity
-        (False, True, False),  # interface vector cell-centered quantity
+        (True, True, True, None),
+        (True, False, True, None),
+        (True, True, False, None),
+        (True, False, False, None),
+        (False, True, True, None),
+        (False, True, False, 1.2),
     ],
 )
+@pytest.mark.parametrize(
+    "grid_fixture", ["grids", "grids_3d"]
+)  # Use both 2D and 3D grid fixtures
 def test_l2_error(
     is_sd: bool,
-    is_scalar: bool,
     is_cc: bool,
+    is_scalar: bool,
     is_relative: bool,
-    grids: list[pp.Grid, pp.MortarGrid],
+    parameter_weight: bool,
+    grid_fixture: str,
+    request: pytest.FixtureRequest,
 ) -> None:
     """Test whether the discrete L2-error is computed correctly.
 
     The test sets arrays of ones as for the true array, and arrays of zeros for the
-    approximate arrays. The absolute l2-error is thus the square root of the sum of
-    the measure of each element (cell_volume when is_cc=True and face_area when
-    is_cc=False) in each grid. The relative l2-error is always 1.0 in all cases.
+    approximate arrays. The absolute l2-error is thus the square root of the sum of the
+    measure of each element (cell_volume when is_cc=True and face_area when is_cc=False)
+    in each grid. The relative l2-error is always 1.0 in all cases.
 
     Parameters:
         is_sd: Whether the error should be evaluated in a subdomain grid. False
             implies evaluation in an interface grid.
-        is_scalar: Whether the array is corresponds to a scalar quantity. False
-            implies a vector quantity.
         is_cc: Whether the array is a cell-centered quantity. False implies a
             face-centered quantity.
-        grids: List of grids. The first element is a two-dimensional subdomain grid,
-            and the second element is an interface grid. See the fixture grids().
+        is_scalar: Whether the array is corresponds to a scalar quantity. False
+            implies a vector quantity.
+        parameter_weight: Weight to be applied to the error. If a scalar is provided,
+            the same weight is applied to all cells or faces. If an array is provided,
+            the weight is broadcasted to each component of the vector quantity. While
+            the resulting error is the same, this design tests two different
+            implementations of the error computation.
```

Comment:
Documentation not consistent with the signature.

Reply:
I updated this documentation, but apparantly forgot to push. Sorry!